Add Events and Hooks Reference

CommanderFoo opened the pull request:
First draft for events and hooks.

PopThosePringles committed:
First draft for events and hooks.

PopThosePringles committed:
Links and semi colon

Someone let a semi colon slip into the code.  :D
Change links to point to the md file.

PopThosePringles committed:
Corrected comment in code

PopThosePringles committed:
Included platform limits link (forum) in rates limit section

StanzillaManticore commented on src/references/events_hooks.md:
```diff
@@ -0,0 +1,433 @@
+!!! tip
+    A **Trigger** will also work in a **Client Context** by setting the **Collision** property to **Force On**.
+
+!!! warn
```
They keyword is `warning`
```suggestion
!!! warning
```

In general, I'm not too much of a fan of having to admonition boxes right after each other, maybe we can have one above the code instead? 

Also, if you don't want to give admonitions a title, you can put the text in quotes and it will collapse for a more compact look like so:

`!!! warning "When a **Vehicle** enters or exits a **Trigger**, it will fire the listener twice."`

![image](https://user-images.githubusercontent.com/58569561/129583612-33ea76ae-1585-46bf-9c12-4e965b0be565.png)

StanzillaManticore commented on src/references/events_hooks.md:
````diff
@@ -0,0 +1,433 @@
+    ```lua linenums="1"
+    -- Script "B"
+
+    local evt = nil
````
I actually prefer more verbose variable names over shortened ones, easier to understand for people. Don't really need the `= nil` here since that is the default.

```suggestion
    local myEvent
```

StanzillaManticore commented on src/references/events_hooks.md:
````diff
@@ -0,0 +1,433 @@
+```lua linenums="1"
+-- A table will be used to store all listeners
+
+local evts = {}
````
```suggestion
local eventsTable = {}
```

or

```suggestion
local myEvents = {}
```

same as above, just a little more verbose name.

PopThosePringles committed:
Changed variables, tips, and warn

Variables are now verbose.
Tip and warning are condensed into single lines, and split apart from each other.

PopThosePringles committed:
Changed event name, conflicted with event listener

PopThosePringles committed:
Comment updated

PopThosePringles committed:
Code consistency

PopThosePringles committed:
Removed low priority sentence, as that isn't really the case

StanzillaManticore committed:
Nits

StanzillaManticore reviewed (changes requested):
Needs a Learn More section, other lgtm!

PopThosePringles committed:
Added Learn more.
